### DISCUSSION TITLE
Is there an cuda_ad_spectral mode?

### DISCUSSION THREAD

**Discussion by lschlieder** [mitsuba-renderer/mitsuba3#437] [Q&A] [answered]
Hey everyone. I started looking into this project recently, and i was wondering if a 'cuda_ad_spectral' mode exists, and why it isn't int the default python package. The documentation says i could compile the source myself with different modes, but does not mention which modes exist. Is this something Mitsuba3 can do, or is ad with spectrum not implemented at the moment (since none of the modes are ad spectral)? If so do i need to complie it myself, or is there some pip magic to make my life easier?

Edit: Ok, after more reading the docs, there seems to be a cuda_ad_spetral mode, that is only accessible by compiling from source. I would love it, if there was some precompiled option available, but atm its just great that it exists at all.

**Comment by njroussel** [comment, marked as answer]
Hi @lschlieder 

This page should have all the information you are looking for: https://mitsuba.readthedocs.io/en/stable/src/key_topics/variants.html

As you can see in the list, `cuda_ad_spectral` does indeed exist. We only ship a few variants with the package PyPI to keep the overall size of the wheel small. 

The only option, currently, to get any other variant than the 4 variants included in the PyPI package is to build Mitsuba yourself. We would like to eventually have variants be their own individual PyPI packages that could be somehow linked to a main mitsuba package, but that would require some work in the current build system.
